src/components/compress/index.tsx

```diff
@@ -85,12 +85,18 @@ interface UpdateImageOptions {
   skipPreprocessing?: boolean;
 }
 
-function processInput(
+async function processInput(
   data: ImageData,
   inputProcessData: InputProcessorState,
   processor: Processor,
 ) {
-  return processor.rotate(data, inputProcessData.rotate);
+  let processedData = data;
+
+  if (inputProcessData.rotate.rotate !== 0) {
+    processedData = await processor.rotate(processedData, inputProcessData.rotate);
+  }
+
+  return processedData;
```

Comment:
I could simplify this to:

```js
if (inputProcessData.rotate.rotate === 0) return data;
return processor.rotate(processedData, inputProcessData.rotate);
```

The current pattern is more like what we have for preprocessors now. I guess the pattern in the PR is a bit premature?

Reply:
Probably compresses to the same output ;)